app/models/item.rb

```diff
@@ -16,4 +16,13 @@ def total_revenue
     invoice_items.sum("quantity * unit_price").to_s.rjust(3, "0").insert(-3, ".")
   end
 
+  def best_date
+    invoices.joins(:transactions, :invoice_items)
+            .where('transactions.result = ?', 'success')
+            .select('max(invoice_items.quantity), invoices.created_at')
+            .group('invoices.created_at')
+            .pluck('invoices.created_at')
+            .first
+            .strftime('%B %d, %Y')
```

Comment:
I think formatting is supposed to be done in the view, it might be better to use the strftime once you are displaying the date on the page.

Reply:
I moved this to the view and I added a model test. Please review and let me know what you think.